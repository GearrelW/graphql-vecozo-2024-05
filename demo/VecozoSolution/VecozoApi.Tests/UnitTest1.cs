using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using Snapshooter.MSTest;
using VecozoApi.Entities;
using VecozoApi.Repositories;

namespace VecozoApi.Tests
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public async Task SchemaDoesNotChangeTest()
        {
            var schema = await new ServiceCollection()
                .AddGraphQL()
                .AddFiltering()
                .AddQueryType<QueryType>()
                .AddMutationType<MutationType>()
                .BuildSchemaAsync();

            schema.ToString().MatchSnapshot();
        }

        [TestMethod]
        public async Task StaticPropertyTest()
        {
            var result = await new ServiceCollection()
                .AddGraphQL()
                .AddFiltering()
                .AddQueryType<QueryType>()
                .AddMutationType<MutationType>()
                .ExecuteRequestAsync("{ andertekstje }");

            result.ToJson().MatchSnapshot();
        }

        [TestMethod]
        public async Task PropertyWithMockingTest()
        {
            // Moq
            // NSubstitute
            // FakeItEasy

            var mockShowRepo = new Mock<IShowRepository>();
            mockShowRepo.Setup(x => x.GetAll()).ReturnsAsync(new List<Show>
            {
                new() { Title = "test1" },
                new() { Title = "test2" }
            });

            var result = await new ServiceCollection()
                .AddGraphQL()
                .AddFiltering()
                .AddQueryType<QueryType>()
                .AddMutationType<MutationType>()
                .Services
                .AddTransient(_ => mockShowRepo.Object)
                .BuildServiceProvider()
                .ExecuteRequestAsync("{ shows { title } }");

            result.ToJson().MatchSnapshot();
        }
    }
}