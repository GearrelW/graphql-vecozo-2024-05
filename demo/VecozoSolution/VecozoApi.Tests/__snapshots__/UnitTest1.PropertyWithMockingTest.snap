{
  "data": {
    "shows": [
      {
        "title": "test1"
      },
      {
        "title": "test2"
      }
    ]
  }
}
